iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java

```diff
@@ -753,12 +812,334 @@ public static void sendCommand(
+    // [START bind_device_to_gateway]
+    createDevice(projectId, cloudRegion, registryName, deviceId);
+
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault().createScoped(CloudIotScopes.all());
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    final CloudIot service =
+        new CloudIot.Builder(GoogleNetHttpTransport.newTrustedTransport(), jsonFactory, init)
+            .setApplicationName(APP_NAME)
+            .build();
+
+    final String registryPath =
+        String.format(
+            "projects/%s/locations/%s/registries/%s", projectId, cloudRegion, registryName);
+
+    BindDeviceToGatewayRequest request = new BindDeviceToGatewayRequest();
+    request.setDeviceId(deviceId);
+    request.setGatewayId(gatewayId);
+
+    BindDeviceToGatewayResponse response =
+        service
+            .projects()
+            .locations()
+            .registries()
+            .bindDeviceToGateway(registryPath, request)
+            .execute();
+
+    System.out.println(String.format("Device bound: %s", response.toPrettyString()));
+    // [END bind_device_to_gateway]
+  }
+
+  public static void unbindDeviceFromGateway(
+      String projectId, String cloudRegion, String registryName, String deviceId, String gatewayId)
+      throws GeneralSecurityException, IOException {
+    // [START unbind_device_from_gateway]
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault().createScoped(CloudIotScopes.all());
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    final CloudIot service =
+        new CloudIot.Builder(GoogleNetHttpTransport.newTrustedTransport(), jsonFactory, init)
+            .setApplicationName(APP_NAME)
+            .build();
+
+    final String registryPath =
+        String.format(
+            "projects/%s/locations/%s/registries/%s", projectId, cloudRegion, registryName);
+
+    UnbindDeviceFromGatewayRequest request = new UnbindDeviceFromGatewayRequest();
+    request.setDeviceId(deviceId);
+    request.setGatewayId(gatewayId);
+
+    UnbindDeviceFromGatewayResponse response =
+        service
+            .projects()
+            .locations()
+            .registries()
+            .unbindDeviceFromGateway(registryPath, request)
+            .execute();
+
+    System.out.println(String.format("Device unbound: %s", response.toPrettyString()));
+    // [END unbind_device_from_gateway]
+  }
+
+  public static void sendDataFromDevice(
+      MqttClient client, String deviceId, String messageType, String data) throws MqttException {
+    // [START send_data_from_bound_device]
+    if (!messageType.equals("events") && !messageType.equals("state")) {
+      System.err.println("Invalid message type, must ether be 'state' or events'");
```

Comment:
Would be nice to include the event type that was received in the error message.

Reply:
OK, I will do that :)